src/main/java/com/uid2/admin/vertx/api/annotations/ApiMethod.java

```diff
@@ -0,0 +1,12 @@
+package com.uid2.admin.vertx.api.annotations;
+
+import io.vertx.core.http.HttpMethod;
+
+public enum ApiMethod {
+    GET(HttpMethod.GET);
```

Comment:
I am not familiar with this, what does this line mean?

Reply:
That's passing HttpMethod.GET to the enum constructor for GET.

If the enum had no constructor, it would just be:
`public enum ApiMethod {
  GET;
}

But because there's a constructor which requires a `HttpMethod` argument, instead of just `GET` you need to pass it the matching `HttpMethod`.

It's because I wanted to use the HTTP method in an attribute, but unfortunately Vert.x's methods aren't compile-time constants 😥 so this enum gives a way to match.

Once someone needs `POST` (or another verb) it should be obvious enough how to add it.

Reply:
Oh, I see, I didn't realise this ApiMethod is an enum! the `HttpMethod.GET` is the vertxMethod that passes into the constructor now it makes sense to me. Thanks for the explanation.